### ISSUE TITLE
No proper error page in case of fatal error when this bundle is activated

### ISSUE THREAD

**Issue by hectorj** [getsentry/sentry-symfony#79]
Hi!

I'm having a small problem with this bundle:

When I activate it, Symfony doesn't handle PHP fatal errors anymore.

Let's say I have a `$this->nonExistingMethod();` in one of my controller. Without this bundle, in prod env, Symfony displays a "pretty" HTML error page.

With this bundle, I get this instead:

![error 500](https://user-images.githubusercontent.com/2427959/31766668-8c66e0b6-b4c8-11e7-9942-d4740e1944fc.png)

My env is:
PHP 7.1
Symfony v3.3.10
sentry/sentry-symfony 0.8.6

**Comment by Jean85** [comment]
I was able to reproduce it on our prod env, sorry for the issue :( I'll try to debug it...

**Comment by hectorj** [comment]
From what I see, [here](https://github.com/symfony/symfony/blob/v3.3.10/src/Symfony/Component/HttpKernel/EventListener/DebugHandlersListener.php#L118) we get a `\Raven_ErrorHandler` instead of a `\Symfony\Component\Debug\ErrorHandler`, so `\Symfony\Component\HttpKernel\HttpKernel::terminateWithException` (instantiated [here](https://github.com/symfony/symfony/blob/v3.3.10/src/Symfony/Component/HttpKernel/EventListener/DebugHandlersListener.php#L105) and usually registered [here](https://github.com/symfony/symfony/blob/v3.3.10/src/Symfony/Component/HttpKernel/EventListener/DebugHandlersListener.php#L123)) is never registered.

That's because `\set_exception_handler` is called by `\Raven_ErrorHandler::registerExceptionHandler` which is called by `\Raven_Client::install` which is called when Symfony instantiates it (following the config [here](https://github.com/getsentry/sentry-symfony/blob/0.8.6/src/Sentry/SentryBundle/Resources/config/services.yml#L6)), which happens when Symfony instantiates `\Sentry\SentryBundle\EventListener\ExceptionListener`, which happens when Symfony gets all the listeners for the `KernelRequest` event. (sorry for the run-on sentence).

Without the Sentry bundle, `\Symfony\Component\Debug\ErrorHandler` is the registered exception handler and `\Symfony\Component\HttpKernel\HttpKernel::terminateWithException` gets registered into it.

I'm still investigating more.

**Comment by hectorj** [comment]
To have the desired behavior, one way is to have `\Raven_ErrorHandler` register itself via `\Symfony\Component\Debug\ErrorHandler::setExceptionHandler` instead of `\set_exception_handler`, and only after `\Symfony\Component\HttpKernel\EventListener\DebugHandlersListener::configure` was executed (so it can get and use the `\Symfony\Component\HttpKernel\HttpKernel::terminateWithException` as its `$this->old_exception_handler`).

Or something like that :thinking:

**Comment by Jean85** [comment]
I've closed #80 because it doesn't seem to fix the issue, and we shouldn't merge it as it is.

As said there, Sentry's and Symfony's error handlers are somehow colliding, creating double reporting (see #48) or losing the custom error page. I currently don't know how to fix this.

**Comment by DonCallisto** [comment]
@Jean85 could [`restore_exception_handler` help there somehow?](http://php.net/manual/en/function.restore-exception-handler.php)

Don't know where to place it btw

**Comment by Jean85** [comment]
Yes the uncertainty here is very high. Symfony's error handler is the one that's bothering me, but we cannot be sure if that is the one immediately before ours: an other bundle could have added his handler in the middle.

I'm trying to set up an example Symfony project to test out better this issue.

**Comment by gondo** [comment]
i've just experienced the same problem.
Symfony 2.8
error triggered via random string (not valid php) placed in action:
```
indexAction(Request $request) {
    random string to trigger error
}
```

**Comment by Jean85** [comment]
I've taken an other step further to recreate the issue locally by replacing the ExceptionController instead of replacing the templates, but still no reproduction.

**Comment by Jean85** [comment]
Closing this as not reproducible. Feel free to reply with a reproducible case and I will reopen it.

**Comment by gondo** [comment]
no worries, we will simply use product what actually works

**Comment by DarthSaracen** [comment]
Had many troubles with duplicated errors sent to Sentry (version 2.0.1). After lots of debugging discovered that there are two subscribed handlers: 
1. Sentry\SentryBundle\EventListener\ExceptionListener from symfony bundle which is great and more customizable
2. Raven_ErrorHandler which is registered by calling \Raven_Client::install.

Solved my problem by rewriting this section in sentry.yaml:

`    sentry.client:
        class: '%sentry.client%'
        arguments: ['%sentry.dsn%', '%sentry.options%']
        public: true`

In that way I've just omitted `
    calls:
      - [install, []]` part from
vendor/sentry/sentry-symfony/src/Resources/config/services.yml

Is there any chances to get some option to turn off this install method? Or maybe I just did something wrong? : \

**Comment by Jean85** [comment]
@DarthSaracen yes, that is _kinda_ wrong. If you do not call `install()`, you will miss any error that is not catched by Symfony for any reason, like a fatal error or a memory limit issue.

Double reporting was a bug of the underlying `sentry-php` client, are you using the latest (1.8.3) version?

**Comment by DarthSaracen** [comment]
@Jean85 yeah, I didn't notice that, I was using 1.8.2 but when upgraded to 1.8.3 the problem is still here.
When Im printing debug backtrace right [here](https://github.com/getsentry/sentry-php/blob/23a717e54bda35de1ee21294e8155073769fbe38/lib/Raven/Client.php#L1020), it prints the following:

`#0  Raven_Client->send() called at [.../vendor/sentry/sentry/lib/Raven/Client.php:898]
#1  Raven_Client->capture() called at [.../vendor/sentry/sentry/lib/Raven/Client.php:650]
#2  Raven_Client->captureException() called at [.../vendor/sentry/sentry/lib/Raven/ErrorHandler.php:78]
#3  Raven_ErrorHandler->handleException() called at [.../vendor/sentry/sentry/lib/Raven/ErrorHandler.php:103]
#4  Raven_ErrorHandler->handleError()
#5  trigger_error() called at [.../src/Controller/Index.php:31]
#6  App\Controller\Index->indexAction() called at [.../vendor/symfony/http-kernel/HttpKernel.php:151]
#7  Symfony\Component\HttpKernel\HttpKernel->handleRaw() called at [.../vendor/symfony/http-kernel/HttpKernel.php:68]
#8  Symfony\Component\HttpKernel\HttpKernel->handle() called at [.../vendor/symfony/http-kernel/Kernel.php:169]
#9  Symfony\Component\HttpKernel\Kernel->handle() called at [.../public/index.php:38]
string(9) "Hey there"

#0  Raven_Client->send() called at [.../vendor/sentry/sentry/lib/Raven/Client.php:898]
#1  Raven_Client->capture() called at [.../vendor/sentry/sentry/lib/Raven/Client.php:650]
#2  Raven_Client->captureException() called at [.../src/Sentry/ExceptionListener.php:40]
#3  App\Sentry\ExceptionListener->onKernelException()
#4  call_user_func() called at [.../vendor/symfony/event-dispatcher/Debug/WrappedListener.php:104]
#5  Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke() called at [.../vendor/symfony/event-dispatcher/EventDispatcher.php:212]
#6  Symfony\Component\EventDispatcher\EventDispatcher->doDispatch() called at [.../vendor/symfony/event-dispatcher/EventDispatcher.php:44]
#7  Symfony\Component\EventDispatcher\EventDispatcher->dispatch() called at [.../vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:139]
#8  Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch() called at [.../vendor/symfony/http-kernel/HttpKernel.php:228]
#9  Symfony\Component\HttpKernel\HttpKernel->handleException() called at [.../vendor/symfony/http-kernel/HttpKernel.php:79]
#10 Symfony\Component\HttpKernel\HttpKernel->handle() called at [.../vendor/symfony/http-kernel/Kernel.php:169]
#11 Symfony\Component\HttpKernel\Kernel->handle() called at [.../public/index.php:38]
string(22) "User Notice: Hey there"`

The error is triggered by trigger_error() function. But when throwing an exception, there's no duplicates

**Comment by Jean85** [comment]
Ok, so it's the usage of `trigger_error`. Can you report this as a separate issue?
